### PR TITLE
cov: add tests for ignite.metrics

### PR DESCRIPTION
**Pull request by ydcjeff**
Part of #1790 

Description: Add tests / Removed unused code for ignite.metrics

Check list:

- [x] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: ignite/metrics/metric.py
Comment (ydcjeff) [inline ignite/metrics/metric.py:481]: These are only applicable to py2

File: ignite/metrics/rouge.py
Comment (ydcjeff) [inline ignite/metrics/rouge.py:189]: This is no need to catch. The same error has been caught above.

**Comment by vfdev-5** [comment]
@ydcjeff, just a small though on saving computational resources while merging master to all PRs. When PR is approved we can update it once and then merge and avoid updating all PRs.

File: tests/ignite/metrics/test_metric.py
Comment (vfdev-5) [inline tests/ignite/metrics/test_metric.py:223]: Let's add however tests like 
```python
# division
    m0_div_m1 = m0 / m1
    m0.update([1, 10, 100])
    m1.update([1, 10, 100])
    assert m0_div_m1.compute() == 0.1
    m1.update([2, 20, 200])
    m0.update([2, 20, 200])
assert m0_div_m1.compute() == 10
```

**Review by vfdev-5 — APPROVED** [review]
Thanks for the PR @ydcjeff !

**Pull request merged**
